Enhance support for very long transversals in a permutation group by forcing down huge index steps through an intermediate subgroup

hulpke opened the pull request:
When computing a transversal in a permutation group force down huge index steps through an intermediate subgroup, as they might lead to creation of infeasibly long bit lists.

This is still lacking a test example that would perform in reasonable time.

Commit:
ENHANCE: Very long transversals

When computing a transversal in a permutation group force down huge index
steps, as they might lead to creation of infeasibly long bit lists.

Incidentally, this allows for Filtering classes in A_15.

ThomasBreuer commented:
I have tried the following. (Probably this is a very naive idea.)
```
gap> BITLIST_LIMIT_TRANSVERSAL:= 59;;
gap> G:= AtlasGroup( "J1" );;
gap> U:= TrivialSubgroup( G );;
gap> RightTransversalPermGroupConstructor( IsRightTransversalPermGroupRep, G, U );
```
Using `Print` statements in the code, one can then see that the new code is entered,
but what happens is that the `while i<=Length(ac) do` loop goes on and on, because `i` gets increased only by `1` in each iteration, but the list `ac` may grow faster.
What am I misunderstanding?

hulpke commented:
BITLIST_LIMIT_TRANSVERSAL should not be smaller than MAX_SIZE_TRANSVERSAL.

fingolfin commented:
@hulpke would it be OK to add this test? We can also do that for you so it's no extra work

ThomasBreuer committed:
add a test for the added code

... and fix an `Assert` call in `FactoredTransversal`,
not all input transversals need have components `group` and `subgroup`.

ThomasBreuer committed:
do not call `AtlasGroup` in testfiles

... and better also do not call `SimpleGroup`,
since it would delegate to `PrimitiveGroup`,
which sooner or later will also be forbidden in testfiles

ThomasBreuer commented:
@hulpke I had to change an `Assert` call because it assumed too much about the internal representations of transversals. Now the check is perhaps too weak. Perhaps you have an idea how to improve it.
